server/src/server.ts

```diff
@@ -0,0 +1,264 @@
+        value = urlPath[1].toLowerCase() + urlPath.substr(2);
+    } else {
+        // other path
+        value = urlPath;
+    }
+    return path.normalize(value);
+}
+
+function validateTextDocument(textDocument: TextDocument): void {
+    console.log('validate ' + textDocument.uri);
+
+    const fileName = getFsPath(textDocument.uri);
+    const baseName = path.basename(fileName);
+
+    const gltfText = textDocument.getText();
+    const folderName = path.resolve(fileName, '..');
+
+    const map = tryGetJsonMap(textDocument);
+    if (!map) {
+        let diagnostics: Diagnostic[] = [getDiagnostic({ message: 'Error parsing JSON document.' }, {})];
+        connection.sendDiagnostics({ uri: textDocument.uri, diagnostics });
+        return;
+    }
+
+    gltfValidator.validateString(baseName, gltfText, (uri) =>
+        new Promise((resolve, reject) => {
+            uri = path.resolve(folderName, uri);
+            fs.readFile(uri, (err, data) => {
+                console.log("Loading external file: " + uri);
+                if (err) {
+                    console.warn("Error: " + err.toString());
+                    reject(err.toString());
+                    return;
+                }
+                resolve(data);
+            });
+        }),
+        {
+            maxIssues: currentSettings.Validation.maxIssues,
+            ignoredIssues: currentSettings.Validation.ignoredIssues,
+            severityOverrides: currentSettings.Validation.severityOverrides
+        }
+    ).then((result) => {
+        let diagnostics: Diagnostic[] = [];
+        if (result.issues && result.issues.messages) {
+            const messages = result.issues.messages;
+            const numMessages = messages.length;
+            for (let i = 0; i < numMessages; ++i) {
+                let info = messages[i];
+                if (info.message) {
+                    diagnostics.push(getDiagnostic(info, map));
+                }
+            }
+
+            if (result.issues.truncated) {
+                diagnostics.push(getDiagnostic({
+                    message: 'VALIDATION ABORTED: Too many messages produced.'
+                }, map));
+            }
+        }
+        // Send the computed diagnostics to VSCode, clearing any old messages.
+        connection.sendDiagnostics({ uri: textDocument.uri, diagnostics });
+    }, (result) => {
+        // Validator's error
+        console.warn('glTF Validator had problems.');
+        console.warn(result);
+        let diagnostics: Diagnostic[] = [getDiagnostic({ message: 'glTF Validator error: ' + result }, {})];
+        connection.sendDiagnostics({ uri: textDocument.uri, diagnostics });
+    });
+}
+
+function getDiagnostic(info: any, map: any) : Diagnostic {
+    let range = {
+        start: { line: 0, character: 0 },
+        end: { line: 0, character: Number.MAX_VALUE }
+    };
+
+    let severity: DiagnosticSeverity = DiagnosticSeverity.Error;
+    switch (info.severity) {
```

Comment:
Local variable `severity` and `switch` statement could be removed, and returned object could be changed to:
```js
return {
    code: info.code,
    severity: info.severity,
    range,
    message: info.message,
    source: 'glTF Validator'
};
```

Reply:
The switch is pretty wordy, but it's a very type-safe conversion and not a performance concern at all.  I prefer to keep the Validator severity numbers separate from the VSCode enum, even though they're the same values.  We don't gain much by coupling them too tightly here.